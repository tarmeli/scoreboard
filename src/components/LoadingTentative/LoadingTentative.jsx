import React from "react";
import Div from "../Element/Div/Div";

const LoadingTentative = ({ isLoading, children }) =>
    isLoading ? <Div>Loading...</Div> : <Div>{children}</Div>;

export default LoadingTentative;
