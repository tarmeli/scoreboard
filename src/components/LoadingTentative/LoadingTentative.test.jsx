import React from "react";
import { mount } from "enzyme";

import LoadingTentative from "./LoadingTentative";

describe("LoadingTentative", () => {
    let component;

    beforeEach(() => {
        component = mount(
            <LoadingTentative isLoading={false}>Some children</LoadingTentative>
        );
    });

    it("renders", () => {
        expect(component.render()).toMatchSnapshot();
    });

    it("when loading, shows loading indicator", () => {
        component = mount(
            <LoadingTentative isLoading={true}>Some children</LoadingTentative>
        );

        expect(component.render()).toMatchSnapshot();
    });
});
