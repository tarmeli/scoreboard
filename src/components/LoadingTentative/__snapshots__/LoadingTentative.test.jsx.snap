// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingTentative renders 1`] = `
<div>
  Some children
</div>
`;

exports[`LoadingTentative when loading, shows loading indicator 1`] = `
<div>
  Loading...
</div>
`;
